modules/vault-cluster/variables.tf

```diff
@@ -76,8 +76,13 @@ variable "allowed_ssh_security_group_ids" {
 }
 
 variable "cluster_tag_key" {
-  description = "Add a tag with this key and the value var.cluster_name to each Instance in the ASG."
-  default     = "Name"
+  description = "Add a tag with this key and the value var.cluster_tag_value to each Instance in the ASG."
+  default     = "vault-servers"
+}
+
+variable "cluster_tag_value" {
+  description = "Add a tag with this value and the key var.cluster_tag_key to each Instance in the ASG."
+  default     = "auto-join"
```

Comment:
Looking at this with fresh eyes, it seems like the approach we took with tags is a little too hard-coded. A better approach may be to take advantage of the [tags](https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html#tags) parameter on ASGs. Something along the lines of:

```hcl
variable "tags" {
  type = "list"
  # Example:
  # default = {
  #   {
  #     key = "foo"
  #     value = "bar"
  #     propagate_at_launch = true  
  #   }
  # }
}

resource "aws_autoscaling_group" "vault" {
  # ...
  tags = "${merge(map("Name", var.cluster_name), var.tags)}"
}
```

This way, users can set any arbitrary tags they want, including overriding the `Name` if they so choose.

Reply:
@brikis98 I agree this is a much better solution. I'll made the changes in my PR.